### PR TITLE
Fix casting error while executing process(1.2.x)

### LINKED ISSUE #22363: [System] Casting issue when there is an error while executing a process
**Description:**
Following error comes when there is an error while executing a process. Due to specific error doesn't handle correctly.
```
error: {ballerina}TypeCastError message=incompatible types: 'error' cannot be cast to 'system:Process|system:InvalidOperationError|system:PermissionError|system:FileSystemError|system:FileNotFoundError'
        at ballerina:nativeExec(system.bal:85)
           ballerina:exec(system.bal:82)
           daneshk.system_exec:main(main.bal:5)
```

**Affected Versions:**
1.2.0

**OS, DB, other environment details and versions:**
All environments

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
Add unhandled error type in the system module

Fixes #22363

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/system/src/main/ballerina/src/system/system_errors.bal [hunk 1]
```diff
@@ -47,5 +47,11 @@ public const FILE_NOT_FOUND_ERROR = "{ballerina/system}FileNotFoundError";
...
 
+# Represents the error code for process execute error.
+public const PROCESS_EXEC_ERROR = "{ballerina/system}ProcessExecError";
+
+# Represents an `ProcessExecError` with a detailed message.
+public type ProcessExecError error<PROCESS_EXEC_ERROR, Detail>;
+
 # Represents System related errors.
-public type Error InvalidOperationError|PermissionError|FileSystemError|FileNotFoundError;
+public type Error InvalidOperationError|PermissionError|FileSystemError|FileNotFoundError|ProcessExecError;
```
Comment (MaryamZi) [inline stdlib/system/src/main/ballerina/src/system/system_errors.bal:57]: Isn't this backward incompatible?

Comment (daneshk) [reply stdlib/system/src/main/ballerina/src/system/system_errors.bal:57]: This error type is something we missed in previous releases. I think it is better to correct this now. There is an issue if the users handled it at a low level.

Comment (daneshk) [reply stdlib/system/src/main/ballerina/src/system/system_errors.bal:57]: Or we will reuse existing error type for patch release and correct it in minor release

Comment (MaryamZi) [reply stdlib/system/src/main/ballerina/src/system/system_errors.bal:57]: Yeah, +1. We'll still have the details right? Only the reason will be incorrect.

Comment (daneshk) [reply stdlib/system/src/main/ballerina/src/system/system_errors.bal]: Yes. fixed in 0e7ea63

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemConstants.java [hunk 2]
```diff
@@ -55,7 +55,7 @@
     static final String PROCESS_FIELD = "ProcessField";
 
     // System error codes
-    public static final String PROCESS_EXEC_ERROR = "{ballerina/system}ProcessExecError";
+    public static final String PROCESS_EXEC_ERROR = "{ballerina/system}InvalidOperationError";
```
Comment (aashikam) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemConstants.java:58]: Are we using this constant anywhere? Shall we rename it?

Comment (daneshk) [reply stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemConstants.java:58]: This is a temporary fix for a patch release. I did it this way to minimize the changes. I introduced a new error type called `ProcessExecError` in the master branch.

**Pull request merged**